contrib/internal/httptrace/httptrace_api_gateway_test.go

```diff
@@ -0,0 +1,196 @@
+package httptrace
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/normalizer"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var appListener *httptest.Server
+var inferredHeaders = map[string]string{
+	"x-dd-proxy":                 "aws-apigateway",
+	"x-dd-proxy-request-time-ms": "1729780025473",
+	"x-dd-proxy-path":            "/test",
+	"x-dd-proxy-httpmethod":      "GET",
+	"x-dd-proxy-domain-name":     "example.com",
+	"x-dd-proxy-stage":           "dev",
+}
+
+// mock the aws server
+func loadTest(t *testing.T) {
+	// Set environment variables
+	t.Setenv("DD_SERVICE", "aws-server")
+	t.Setenv("DD_TRACE_INFERRED_PROXY_SERVICES_ENABLED", "true")
+
+	// set up http server
+	mux := http.NewServeMux()
+
+	// set routes
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/error" {
+			w.WriteHeader(http.StatusInternalServerError)
+			_ = json.NewEncoder(w).Encode(map[string]string{"message": "ERROR"})
+		} else {
+			w.WriteHeader(http.StatusOK)
+			_ = json.NewEncoder(w).Encode(map[string]string{"message": "OK"})
+		}
+	})
+	appListener = httptest.NewServer(mux)
+
+}
+func cleanupTest() {
+	// close server
+	if appListener != nil {
+		appListener.Close()
+	}
+}
+
+func TestInferredProxySpans(t *testing.T) {
+
+	t.Run("should create parent and child spans for a 200", func(t *testing.T) {
+		mt := mocktracer.Start()
+		defer mt.Stop()
+		loadTest(t)
+		defer cleanupTest()
+
+		client := &http.Client{}
+		req, err := http.NewRequest("GET", fmt.Sprintf("%s/", appListener.URL), nil)
+
+		assert := assert.New(t)
+		assert.NoError(err)
+
+		for k, v := range inferredHeaders {
+			req.Header.Set(k, v)
+		}
+
+		sp, _ := StartRequestSpan(req)
+		resp, err := client.Do(req)
+		FinishRequestSpan(sp, resp.StatusCode, nil)
+
+		spans := mt.FinishedSpans()
+
+		assert.NoError(err)
+		assert.Equal(http.StatusOK, resp.StatusCode)
+
+		assert.Equal(2, len(spans))
+		gateway_span := spans[0]
+		web_req_span := spans[1]
+		assert.Equal("aws.apigateway", gateway_span.OperationName())
+		assert.Equal("http.request", web_req_span.OperationName())
+		assert.True(web_req_span.ParentID() == gateway_span.SpanID())
+		for _, arg := range inferredHeaders {
+			header, tag := normalizer.HeaderTag(arg)
+			gateway_span_tags, ok := gateway_span.Tags()[tag]
+			if !ok {
+				gateway_span_tags = ""
+			}
+			assert.Equal(strings.Join(req.Header.Values(header), ","), gateway_span_tags)
```

Comment:
Recommend something like this:

```
for _, arg := range inferredHeaders {
    header, tag := normalizer.HeaderTag(arg)
    gateway_span_tags, exists := gateway_span.Tags()[tag]

    // Default to an empty string if the tag doesn't exist
    if !exists {
        gateway_span_tags = ""
    }

    // Compare the expected and actual values
    expected := strings.Join(req.Header.Values(header), ",")
    assert.Equal(expected, gateway_span_tags)
}
```

Makes things a bit more readable especially the assert statement

Reply:
I agree, thank you for the suggestion!